src/mca/pcompress/zlib/compress_zlib.c

```diff
@@ -117,7 +119,7 @@ static bool zlib_compress(const uint8_t *inbytes, size_t inlen, uint8_t **outbyt
     *outlen = len2;
 
     /* fold the uncompressed length into the buffer */
-    memcpy(ptr, &inlen, sizeof(uint32_t));
+    memcpy(ptr, &len3, sizeof(uint32_t));
```

Comment:
I can see here where the switch from a `size_t` to `uint32_t` would make a difference due to alignment within the memory space.

Reply:
A bigger problem would be with big endian systems which would only copy 0 bits. Not sure if these systems are actually still around.

Reply:
A few - not many, and on their way out. Still, best to be correct - thanks for catching it!